homeassistant/components/cast/media_player.py

```diff
@@ -339,6 +340,44 @@ def new_cast_status(self, cast_status):
 
     def new_media_status(self, media_status):
         """Handle updates of the media status."""
+        if (
+            media_status
+            and media_status.player_is_idle
+            and media_status.idle_reason == "ERROR"
+        ):
+            tts_base_url = self.hass.data["tts_base_url"]
```

Comment:
`cast` integration shouldn't access `hass.data` from other integrations. If we need to access it's data, the `tts` integration will need to add a method to `__init__.py` that cast can access.

Reply:
Also this would blow up if `tts` was not loaded.

Reply:
OK, rewritten a bit now.